app/models/user_model.ml

```diff
@@ -85,6 +90,31 @@ let register =
     Db.exec query (name, email, password) |> Lib_db.try_
 ;;
 
+let if_name_empty name user = match name with
+  | "" -> user.name
+  | new_name -> new_name
+
+let if_email_empty email user = match email with
+  | "" -> user.email
+  | new_email -> new_email
```

Comment:
My mistake, it is done into the formlet part! Sorry. I'm not sure that this is the right way to do that but (I strongly prefer having it into the validation part) but I have no real strong opposition. And with your approach, you can keep the validation formlet pure, which seems desireable.

Reply:
(So you can close it if you cannot find a better way to handle that)

Reply:
I think you are completely right @gr-im, it is better to consider that the result of the _formlet_ already contains all the right data to reduce the need for "non-SQL" logic in the insert/modify process. It is possible to leave the validator "pure" by providing the user (which should be, I think, already available because `provide_user`). An example "a little pseudo-coded":

```ocaml
type update_preference_form =
 { preference_name : string
 ; preference_email : string }
 
 let replace_if_blank replacement subject = 
   let real_subject = String.trim subject with in
   if String.(equal real_subject empty) then replacement
   else real_subject
 
let validate_preferences_update ?(name_field = "user_name") ?(email_field = "user_email") user =
  let open Lib_form in
  let formlet s = 
    let+ name = required s name_field (is_string $ replace_if_blank user.name)
    and+ email = required s email_field ((is_email <|> (is_string $ replace_if_blank user.email) $ normalize_name) in
   { preference_name = name
   ; preference_email = email } 
 in run ~name:"User.preferences" formlet
```

the code was not tested and written in a hurry, but I guess you get the idea?

Reply:
Yes, obviously better than mine, thanks.